Add @task.kubernetes decorator section (#2104)
astronomer/docs: learn/kubepod-operator.md

- `pod_template_file`: The path to a Pod template file.
- `full_pod_spec`: A complete Pod configuration formatted as a Python `k8s` object.

> There are many other arguments that can be used to configure the Pod and pass information to the Docker image. For a list of the available KubernetesPodOperator arguments, see the [KubernetesPodOperator source code](https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/pod.py).

Reviewer: ```suggestion
You can also use many other arguments to configure the Pod and pass information to the Docker image. For a list of the available KubernetesPodOperator arguments, see the [KubernetesPodOperator source code](https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/pod.py).
```

The components of the connection can also be set or overwritten at the task level by using the arguments `config_file` (to specify the path to the `KubeConfig` file) and `cluster_context`. Setting these parameters in `airflow.cfg` has been deprecated.

## Use the @task.kubernetes decorator

> The `@task.kubernetes` decorator was added in Airflow 2.4 and provides an alternative to using the traditional KubernetesPodOperator when running Python scripts in within a separate Kubernetes Pod. Note that the Docker image provided to the `@task.kubernetes` decorator must support the execution of Python scripts.

Reviewer: ```suggestion
The `@task.kubernetes` decorator was added in Airflow 2.4 and provides an alternative to the traditional KubernetesPodOperator when you run Python scripts in a separate Kubernetes Pod. The Docker image provided to the `@task.kubernetes` decorator must support executing Python scripts.
```

> XComs can be passed to the Python script running within the dedicated Kubernetes pod as with regular `@task` decorated functions. The value returned by the decorated function will be pushed to XCom if `do_xcom_push` is set to `True` in the decorator parameters. You can learn more about decorators in the [Introduction to Airflow decorators](airflow-decorators.md) guide.

Reviewer: ```suggestion
Like regular `@task` decorated functions, XComs can be passed to the Python script running in the dedicated Kubernetes pod. If `do_xcom_push` is set to `True` in the decorator parameters, the value returned by the decorated function is pushed to XCom. You can learn more about decorators in the [Introduction to Airflow decorators](airflow-decorators.md) guide.
```

> Astronomer recommends using the `@task.kubernetes` decorator over the KubernetesPodOperator when using XCom with Python scripts inside a dedicated Kubernetes pod.

Reviewer: ```suggestion
Astronomer recommends using the `@task.kubernetes` decorator instead of the KubernetesPodOperator when using XCom with Python scripts in a dedicated Kubernetes pod.
```
